rules/enforce-single-end-return.js

```diff
@@ -32,42 +33,46 @@ module.exports = {
             if (body.length === 0) {
                 return;
             }
+            let function_name = node.name ? `"${node.name}"` : 'Fallback function';
             for (let statement of body.slice(0, -1)) {
                 if (statement['type'] === 'ReturnStatement') {
                     context.report({
                         node: node,
-                        message: 'Return statement not at end of function',
+                        message: `${function_name}: Return statement not at end of function`,
                     });
                 }
             }
             let last_statement = body[body.length - 1];
             if (last_statement['type'] !== 'ReturnStatement') {
                 context.report({
                     node: node,
-                    message: 'Missing return statement at end of function'
+                    message: `${function_name}: Missing return statement at end of function`
                 });
             }
         }
 
-        function inspectReturnDeclaration(emitted) {
+        function inspectBlockStatement(emitted) {
             var node = emitted.node;
 
             if (emitted.exit) {
                 return;
             }
 
-            let parent_node = context.getSourceCode().getParent(node);
-            if (parent_node['type'] !== 'FunctionDeclaration') {
-                context.report({
-                    node: node,
-                    message: 'Missing return statement at end of function'
-                });
+            if (node.parent.type !== 'FunctionDeclaration') {
+                for (let statement of node.body) {
+                    if (statement.type === 'ReturnStatement') {
+                        context.report({
+                            node: node,
+                            message: 'Missing return statement at end of function'
```

Comment:
The message seems misleading (because we just saw a return stmt inside a block and declared that there is no return at the end. There could also be one at the end)
```
function foo() {
    blah (123);
   if (false) { return; }
   bar ("Hello");
   return 2;
}
```

Please change message to `Function <FUNC NAME> should only have a single return statement at the end.`
Or am I missing something?